pkg/cloud/aws/services/ec2/subnets.go

```diff
@@ -113,24 +117,31 @@ LoopExisting:
 	}
 
 	// Proceed to create the rest of the subnets that don't have an ID.
-	for _, subnet := range subnets {
-		if subnet.ID != "" {
-			continue
-		}
+	if !s.scope.VPC().IsProvided() {
```

Comment:
[nit, non-blocking] The rest of the code seems to check provided first then return, why did this flip logic? I'd have expected to see

```go
if s.scope.VPC().IsProvided() {
    klog.V(2).Infof("Provided subnets: %s", subnets)
    return nil
}
```

Reply:
Here we want to describe the subnets, the subnet-id is optional from a user perspective, while the VPC is required. The other components return early because they don't need to be described or validated at this point.